src/utils/bibModelPreprocessing.ts

```diff
@@ -0,0 +1,110 @@
+import type { SearchResult } from "../types/searchTypes"
+
+const getGroupedNotes = (bib: SearchResult) => {
+  const note = bib?.note?.length ? bib.note : null
+  let notesGroupedByNoteType = {}
+
+  /**
+   * getNoteType(note)
+   * Construct label for a note by adding the word 'Note'
+   *
+   * @param {object} note
+   * @return {string}
+   */
+  const getNoteType = (note) => {
+    const type = note.noteType || ""
+    return type.toLowerCase().includes("note") ? type : `${type} (note)`
+  }
+
+  if (!note) {
+    return notesGroupedByNoteType
+  }
+  // Make sure we have at least one note
+  if (note && Array.isArray(note)) {
+    // Group notes by noteType:
+    notesGroupedByNoteType = note
+      // Make sure all notes are blanknodes:
+      .filter((note) => typeof note === "object")
+      .reduce((groups, note) => {
+        const noteType = getNoteType(note)
+        if (!groups[noteType]) {
+          groups[noteType] = []
+        }
+        groups[noteType].push(note.prefLabel)
+        return groups
+      }, {})
+  }
+
+  return notesGroupedByNoteType
+}
+
+/**
+ * combineMatching(el1, el2)
+ * Combines properties from matching (i.e. parallel) elements as necessary
+ * Right now, this is only needed to add the 'noteType' in case of parallel notes
+ * @param {string} el1
+ * @param {object} el2
+ * @return {object}
+ */
+
+const combineMatching = (el1, el2) =>
+  el2 && el2.noteType
+    ? { noteType: el2.noteType, "@type": el2["@type"], prefLabel: el1 }
+    : el1
+
+/**
+ * interleaveParallel(arr1, arr2)
+ * Given two arrays, returns the elements interleaved, with falsey elements removed.
+ * Also combines data from matching elements when necessary.
+ * Example: interleaveParallel ([1, 2, null, 3], [5,6,7,8,9]) =>
+ * [1,5,2,6,7,3,8,9].
+ * Assumes that arr2 is at least as long as arr1.
+ *
+ * @param {array} arr1
```

Comment:
Don't think we need to be using JS doc type annotations for this project since we're using Typescript, but let me know if you disagree.

Reply:
True! this was just an artifact from DFE.